### PR TITLE
Compat - add project infinite compat

### PR DESCRIPTION
**Pull request by alganthe**
**When merged this pull request will:**
- Add compat for the TRG42, MK1, falkor and RK95

### CODE REVIEW

File: optionals/bnae_falkor_compat_ace/CfgWeapons.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Mode_SemiAuto;
+class asdg_OpticRail;
+class asdg_OpticRail1913: asdg_OpticRail {};
+
+class CfgWeapons {
+    class Rifle_Base_F;
+
```
Comment (jonpas) [inline optionals/bnae_falkor_compat_ace/CfgWeapons.hpp]:
```suggestion
```

File: optionals/bnae_falkor_compat_ace/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# About
+
+Add [ACE3](https://github.com/acemod/ACE3) compatibility for [bnae falkor](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```
Comment (jonpas) [inline optionals/bnae_falkor_compat_ace/README.md]:
```suggestion
Adds [ACE3](https://github.com/acemod/ACE3) Compatibility for [Bnae's Falkor Petra](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```

File: optionals/bnae_falkor_compat_ace/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+### Authors
```
Comment (jonpas) [inline optionals/bnae_falkor_compat_ace/README.md]: Add whomever was involved (brainslush? yourself etc.).

File: optionals/bnae_falkor_compat_ace/config.cpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#include "script_component.hpp"
+
+class CfgPatches {
+    class ADDON {
+        units[] = {};
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {"tac_main", "bnae_falkor", "bnae_core"};
```
Comment (jonpas) [inline optionals/bnae_falkor_compat_ace/config.cpp]: Does `bnae_falkor` not already require `bnae_core`? O.o

File: optionals/bnae_falkor_compat_ace/script_component.hpp [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#define COMPONENT bnae_falkor_compat_ace
+#define COMPONENT_BEAUTIFIED BNAE Falkor ACE3 Compat
```
Comment (jonpas) [inline optionals/bnae_falkor_compat_ace/script_component.hpp]:
```suggestion
#define COMPONENT_BEAUTIFIED Bnae Falkor ACE3 Compatibility
```

File: optionals/bnae_mk1_compat_ace/CfgMagazines.hpp [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class CfgMagazines {
+    class CA_Magazine;
+
```
Comment (jonpas) [inline optionals/bnae_mk1_compat_ace/CfgMagazines.hpp]:
```suggestion
```

File: optionals/bnae_mk1_compat_ace/CfgWeapons.hpp [hunk 7]
```diff
@@ -0,0 +1,15 @@
+class Mode_SemiAuto;
+
+class CfgWeapons {
+    class Rifle_Long_Base_F;
+
```
Comment (jonpas) [inline optionals/bnae_mk1_compat_ace/CfgWeapons.hpp]:
```suggestion
```

File: optionals/bnae_mk1_compat_ace/CfgWeapons.hpp [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    class bnae_mk1_base: Rifle_Long_Base_F {
+        initSpeed = -1.0;
+        ACE_barrelTwist = 254.0;
+        ACE_barrelLength = 640.08;
+        ACE_RailHeightAboveBore = 1.8;
+        class Single: Mode_SemiAuto {
```
Comment (jonpas) [inline optionals/bnae_mk1_compat_ace/CfgWeapons.hpp]: Empty line above.

File: optionals/bnae_falkor_compat_ace/CfgWeapons.hpp [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        class WeaponSlotsInfo;
+    };
+
+    // https://www.falkordefense.com/rifles/petra/
+    class bnae_falkor_base: Rifle_Long_Base_F {
+        initSpeed = -1.0;
+        ACE_barrelTwist = 254.0; // 1:10"
+        ACE_barrelLength = 508.0; // 20", the new version has a 22" barrel but the 3D model is the old version with a 20" barrel
+        ACE_RailHeightAboveBore = 3.0;
+        class WeaponSlotsInfo: WeaponSlotsInfo {
```
Comment (jonpas) [inline optionals/bnae_falkor_compat_ace/CfgWeapons.hpp]: Empty line above.

File: optionals/bnae_falkor_compat_ace/CfgWeapons.hpp [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+            class asdg_OpticRail_bnae: asdg_OpticRail1913 {};
+        };
+        class Single: Mode_SemiAuto {
```
Comment (jonpas) [inline optionals/bnae_falkor_compat_ace/CfgWeapons.hpp]: Empty line above.

File: optionals/bnae_mk1_compat_ace/README.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+# About
+
+Add [ACE3](https://github.com/acemod/ACE3) compatibility for [bnae Mk1](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```
Comment (jonpas) [inline optionals/bnae_mk1_compat_ace/README.md]:
```suggestion
Adds [ACE3](https://github.com/acemod/ACE3) Compatibility for [Bnae's Lee-Enfield MK.I](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```

File: optionals/bnae_mk1_compat_ace/script_component.hpp [hunk 12]
```diff
@@ -0,0 +1,17 @@
+#define COMPONENT bnae_mk1_compat_ace
+#define COMPONENT_BEAUTIFIED BNAE Mk1 ACE3 Compat
```
Comment (jonpas) [inline optionals/bnae_mk1_compat_ace/script_component.hpp]:
```suggestion
#define COMPONENT_BEAUTIFIED Bnae MK1 ACE3 Compatibility
```

File: optionals/bnae_rk95_compat_ace/CfgMagazines.hpp [hunk 13]
```diff
@@ -0,0 +1,20 @@
+class CfgMagazines {
+    class CA_Magazine;
+
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/CfgMagazines.hpp]:
```suggestion
```

File: optionals/bnae_rk95_compat_ace/CfgWeapons.hpp [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+    class Rifle;
+    class Rifle_Base_F: Rifle {
+        class WeaponSlotsInfo;
+    };
+
+    // http://guns.connect.fi/gow/M95.html
+    class bnae_rk95_base: Rifle_Base_F {
+        ACE_barrelTwist = 240.0;
+        ACE_barrelLength = 420.0;
+        class Single: Mode_SemiAuto {
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/CfgWeapons.hpp]: Empty line above.

File: optionals/bnae_rk95_compat_ace/CfgWeapons.hpp [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+            dispersion = 0.00044; // 1.5 MOA; default: 0.00058 = 2 MOA
+        };
+        class FullAuto: Mode_FullAuto {
+            dispersion = 0.00044; // 1.5 MOA; default: 0.00058 = 2 MOA
+        };
+    };
+
+    class bnae_rk95_virtual: bnae_rk95_base {};
+
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/CfgWeapons.hpp]:
```suggestion
```

File: optionals/bnae_rk95_compat_ace/CfgWeapons.hpp [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+        ACE_barrelLength = 420.0;
+        class Single: Mode_SemiAuto {
+            dispersion = 0.00044; // 1.5 MOA; default: 0.00058 = 2 MOA
+        };
+        class FullAuto: Mode_FullAuto {
+            dispersion = 0.00044; // 1.5 MOA; default: 0.00058 = 2 MOA
+        };
+    };
+
+    class bnae_rk95_virtual: bnae_rk95_base {};
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/CfgWeapons.hpp]:
```suggestion
    class bnae_rk95_virtual;
```
Just this should work, less chance to break inheritance.

File: optionals/bnae_rk95_compat_ace/CfgWeapons.hpp [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+
+    class bnae_rk95r_virtual: bnae_rk95_virtual {
+        ACE_RailHeightAboveBore = 14.0;
+        class WeaponSlotsInfo: WeaponSlotsInfo {
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/CfgWeapons.hpp]: Empty line above.

File: optionals/bnae_rk95_compat_ace/README.md [hunk 18]
```diff
@@ -0,0 +1,5 @@
+# About
+
+Add [ACE3](https://github.com/acemod/ACE3) compatibility for [bnae rk95](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/README.md]:
```suggestion
Adds [ACE3](https://github.com/acemod/ACE3) Compatibility for [Bnae's Sako RK 95](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```

File: optionals/bnae_rk95_compat_ace/CfgWeapons.hpp [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+            class asdg_OpticRail_bnae: asdg_OpticRail1913 {};
+        };
+    };
+
+    class ItemCore;
+    class InventoryMuzzleItem_Base_F;
+
+    // ACE3 values: https://github.com/acemod/ACE3/blob/master/addons/ballistics/CfgWeapons.hpp#L643
+    class bnae_suppressor_v2_base: ItemCore {
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/CfgWeapons.hpp]: Are those in the same PBO?

File: optionals/bnae_rk95_compat_ace/script_component.hpp [hunk 20]
```diff
@@ -0,0 +1,17 @@
+#define COMPONENT bnae_rk95_compat_ace
+#define COMPONENT_BEAUTIFIED BNAE rk95 ACE3 Compat
```
Comment (jonpas) [inline optionals/bnae_rk95_compat_ace/script_component.hpp]:
```suggestion
#define COMPONENT_BEAUTIFIED Bnae RK95 ACE3 Compatibility
```

File: optionals/bnae_trg42_compat_ace/CfgWeapons.hpp [hunk 21]
```diff
@@ -0,0 +1,94 @@
+class Mode_SemiAuto;
+class asdg_OpticRail;
+
+class asdg_OpticRail1913: asdg_OpticRail {};
+
+class CfgWeapons {
+    class Rifle_Base_F;
+
```
Comment (jonpas) [inline optionals/bnae_trg42_compat_ace/CfgWeapons.hpp]:
```suggestion
```

File: optionals/bnae_trg42_compat_ace/CfgWeapons.hpp [hunk 22]
```diff
@@ -0,0 +1,94 @@
+class Mode_SemiAuto;
+class asdg_OpticRail;
+
+class asdg_OpticRail1913: asdg_OpticRail {};
```
Comment (jonpas) [inline optionals/bnae_trg42_compat_ace/CfgWeapons.hpp]: Why do we change inheritance of that? It already inherits, just include it normally.

File: optionals/bnae_trg42_compat_ace/CfgWeapons.hpp [hunk 23]
```diff
@@ -0,0 +1,94 @@
+class Mode_SemiAuto;
+class asdg_OpticRail;
+
```
Comment (jonpas) [inline optionals/bnae_trg42_compat_ace/CfgWeapons.hpp]:
```suggestion
```

File: optionals/bnae_trg42_compat_ace/CfgWeapons.hpp [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+        class WeaponSlotsInfo;
+    };
+
+    // http://cdn1.sako.fi/sites/default/files/TRG-42_Tech_specs.pdf
+    class bnae_trg42_base: Rifle_Long_Base_F {
+        initSpeed = -1.0;
+        ACE_barrelTwist = 254.0; // 1:10"
+        ACE_barrelLength = 690.0; // 27 1/8"
+        ACE_RailHeightAboveBore = 1.41739;
+        class WeaponSlotsInfo: WeaponSlotsInfo {
```
Comment (jonpas) [inline optionals/bnae_trg42_compat_ace/CfgWeapons.hpp]: Empty line above.

File: optionals/bnae_trg42_compat_ace/CfgWeapons.hpp [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+            class asdg_OpticRail_bnae: asdg_OpticRail1913 {};
+        };
+        class Single: Mode_SemiAuto {
```
Comment (jonpas) [inline optionals/bnae_trg42_compat_ace/CfgWeapons.hpp]: Empty line above.

File: optionals/bnae_trg42_compat_ace/README.md [hunk 26]
```diff
@@ -0,0 +1,5 @@
+# About
+
+Add [ACE3](https://github.com/acemod/ACE3) compatibility for [bnae trg42](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```
Comment (jonpas) [inline optionals/bnae_trg42_compat_ace/README.md]:
```suggestion
Adds [ACE3](https://github.com/acemod/ACE3) Compatibility for [Bnae's Sako TRG-42](https://forums.bohemia.net/forums/topic/196809-project-infinite-v10/).
```

File: optionals/bnae_trg42_compat_ace/script_component.hpp [hunk 27]
```diff
@@ -0,0 +1,17 @@
+#define COMPONENT bnae_trg42_compat_ace
+#define COMPONENT_BEAUTIFIED BNAE trg42 ACE3 Compat
```
Comment (jonpas) [inline optionals/bnae_trg42_compat_ace/script_component.hpp]:
```suggestion
#define COMPONENT_BEAUTIFIED Bnae TRG42 ACE3 Compatibility
```

File: optionals/bnae_falkor_compat_ace/config.cpp (hunk 4 above)
Comment (alganthe) [reply optionals/bnae_falkor_compat_ace/config.cpp]: not sure, that's why I left it as is.

**Pull request merged**